main.go

```diff
@@ -104,15 +115,71 @@ func initLogging() (*os.File, error) {
 	return file, nil
 }
 
+// extension specific PublicSettings
+type PublicSettings struct {
+	Script   string   `json:"script"`
+	FileURLs []string `json:"fileUris"`
+}
+
+// extension specific ProtectedSettings
+type ProtectedSettings struct {
+	SecretString       string   `json:"secretString"`
+	SecretScript       string   `json:"secretScript"`
+	FileURLs           []string `json:"fileUris"`
+	StorageAccountName string   `json:"storageAccountName"`
+	StorageAccountKey  string   `json:"storageAccountKey"`
+}
+
 func main() {
-	// TODO Add more robust error handling
 	file, err := initLogging()
 	if err != nil {
 		fmt.Println("Error opening the provided logfile.")
-		os.Exit(1)
+		os.Exit(-1)
 	}
 	defer file.Close()
 
+	extensionMrseq, environmentMrseq, err := sequence.GetMostRecentSequenceNumber()
+	if err != nil {
+		errorLogger.Printf("%+v", err)
+		os.Exit(-1)
+	}
+	infoLogger.Printf("Extension MrSeq: %d, Environment MrSeq: %d", extensionMrseq, environmentMrseq)
+
+	shouldRun := sequence.ShouldBeProcessed(extensionMrseq, environmentMrseq)
+	if !shouldRun {
+		errorLogger.Printf("environment mrseq has already been processed by extension (environment mrseq : %v, extension mrseq : %v)\n", environmentMrseq, extensionMrseq)
+		os.Exit(-1)
+	}
+	infoLogger.Printf("Extension should run: %t", shouldRun)
+
+	err = sequence.SetExtensionMostRecentSequenceNumber(environmentMrseq)
+	if err != nil {
+		errorLogger.Printf("%+v", err)
+		os.Exit(-1)
+	}
+
+	err = status.ReportTransitioning(environmentMrseq, "install", "installation in progress")
```

Comment:
Its awesome you're able to parse the settings now! Just FYI, this logic would need to be changed eventually, like all this would have to be for specific functions (like reporting install for install command, and so on). 
For the purpose of this PR it looks good though.